src/6model/reprs/MVMSpeshCandidate.c

```diff
@@ -93,13 +103,40 @@ static void deserialize_stable_size(MVMThreadContext *tc, MVMSTable *st, MVMSeri
 /* Calculates the non-GC-managed memory we hold on to. */
 static MVMuint64 unmanaged_size(MVMThreadContext *tc, MVMSTable *st, void *data) {
     MVMSpeshCandidateBody *body = (MVMSpeshCandidateBody *)data;
-    return body->bytecode_size +
-           body->num_handlers * sizeof(MVMFrameHandler) +
-           body->num_deopts * sizeof(MVMint32) +
-           sizeof(MVMSpeshPEADeopt) +
-           body->env_size +
-           body->work_size +
-           (body->jitcode ? body->jitcode->size : 0);
+    MVMuint64 size = 0;
+
+    size += body->bytecode_size;
+
+    size += sizeof(MVMFrameHandler) * body->num_handlers;
+
+    size += sizeof(MVMCollectable *) * body->num_spesh_slots;
+
+    size += sizeof(MVMint32) * body->num_deopts;
+
+    size += sizeof(MVMSpeshInline) * body->num_inlines;
+
+    size += sizeof(MVMuint16) * (body->num_locals + body->num_lexicals);
+
+    /* XXX probably don't need to measure the bytecode size here,
+     * as it's probably just a pointer to the same bytecode we have in
+     * the static frame anyway. */
```

Comment:
We do need to measure it here, I think, as its the specialized bytecode; the static frame is not copied per specialization. So this comment can go.